hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindSourceGreedy.java

```diff
@@ -171,6 +172,10 @@ public boolean canSizeLeaveSource(DatanodeDetails source, long size) {
             source.getUuidString(), lowerLimit);
         return false;
       }
+      if (size <= 0) {
+        LOG.debug("{} bytes container cannot leave datanode {}", size, source.getUuidString());
+        return false;
+      }
```

Comment:
We could move this check to the top (before line 161) as this would avoid making the method calls in the above 2 checks when it is not needed. (the comment above can also be changed accordingly)